tests/amd_migraphx_test/resnet50/migraphx_node_test.cpp

```diff
@@ -0,0 +1,212 @@
+    vx_size output_num_of_dims;
+    vx_enum output_data_format;
+    vx_size output_dims[4];
+    vx_size stride[4];
+    vx_map_id map_id;
+    void *ptr = nullptr;
+    vx_status status = 0;
+    migraphx::program prog;
+
+    //imagenet label file
+    std::string labelText[1000];
+    std::string labelFileName = ("../labels.txt");
+
+    std::string line;
+    std::ifstream out(labelFileName);
+    if(!out) {
+      std::cout << "label file failed to open" << std::endl;
+      return -1; 
+    }
+    int lineNum = 0;
+    while(getline(out, line)) {
+        labelText[lineNum] = line;
+        lineNum++;
+    }
+    out.close();
+
+    status = amdMIGraphXcompile(modelFileName.c_str(), &prog,
+    &input_num_of_dims, input_dims, &input_data_format,
+    &output_num_of_dims, output_dims, &output_data_format, false, false);
+
+    if (status) {
+        std::cerr << "ERROR: amdMIGraphXcompile failed " << std::endl;
+        return status;
+    }
+
+    input_tensor = vxCreateTensor(context, input_num_of_dims, input_dims, input_data_format, 0);
+    output_tensor = vxCreateTensor(context, output_num_of_dims, output_dims, output_data_format, 0);
+
+    //read an image and resize to correct dimensions -- opencv imread()
+    cv::Mat input_image, input_image_224x224;
+    input_image = cv::imread(imageFileName);
+    
+    //resizing
+    int input_width = input_image.size().width;
+    int input_height = input_image.size().height;
+    if(input_height > input_width) {
+      int dif = input_height - input_width;
+      int bar = floor(dif / 2);
+      cv::Range rows((bar + (dif % 2)), (input_height - bar));
+      cv::Range cols(0, input_width);
+      cv::Mat square = input_image(rows, cols);
+      cv::resize(square, input_image_224x224, cv::Size(224,224));
+    } else if(input_width > input_height) {
+      int dif = input_width - input_height;
+      int bar = floor(dif / 2);
+      cv::Range rows(0, input_height);
+      cv::Range cols((bar + (dif % 2)), (input_width - bar));
+      cv::Mat square = input_image(rows, cols);
+      cv::resize(square, input_image_224x224, cv::Size(224,224));
+    } else {
+        cv::resize(input_image, input_image_224x224, cv::Size(224,224));
+    }
+
+    //preprocess
+    cv::Mat RGB_input_image;
+    cv::cvtColor(input_image_224x224, RGB_input_image, cv::COLOR_BGR2RGB);  // cv::imread reads the image in order BGR. SO need to convert
+    
+    int rows = RGB_input_image.rows; int cols = RGB_input_image.cols; 
+    int total = RGB_input_image.total() * RGB_input_image.channels();
+    unsigned char *input_image_vector = (RGB_input_image.data);
+
+    float *buf = (float *)malloc(total*sizeof(float));
+    float *R = buf;
+    float *G = R + rows*cols;
+    float *B = G + rows*cols;
+
+    float mean_vec[3] = {0.485, 0.456, 0.406};
+    float stddev_vec[3] = {0.229, 0.224, 0.225};    
+    float preproc_mul[3] = { 1/(255*stddev_vec[0]), 1/(255*stddev_vec[1]), 1/(255*stddev_vec[2])};
```

Comment:
formatting: please use one space before and one space after arithmetic operators in this line and the rest of this file.